[WIP] 'What is quantum computing?' webpage (#614)
PennyLaneAI/qml: what-is-quantum-computing.rst

> .. role:: html(raw)

Reviewer: There are two unused images that seem to have snuck into this PR :thinking: 
Be good to remove them if not needed
(sorry can't leave comment on the images directly; they're the same ones as in QChem page)

Author: Yeah, probably the same issue. I'll fix it.

Reviewer 2: @ikurecic has this been resolved?

Reviewer 2: perfect, thanks!

   :format: html

What Is Quantum Computing?
==========================

.. meta::

>    :property="og:description": Quantum computing is a research area that extends the set of physical laws classical computers operate on by accessing quantum aspects of the physical world, opening up new ways of processing information.
> 
>    :property="og:image": https://pennylane.ai/qml/_static/whatisqc/quantum_computer.svg

Reviewer 2: ```suggestion
   :property="og:description": Quantum computing is a research area that extends the set of physical laws classical computers operate on by accessing quantum aspects of the physical world, opening up new ways of processing information.
   :property="og:image": https://pennylane.ai/qml/_static/whatisqc/quantum_computer.svg
```

Quantum computing is a research area that extends the set of physical laws classical computers operate on by 
accessing **quantum aspects of the physical world**, opening up new ways of processing information.
As a highly interdisciplinary field of quantum science and technology, quantum computing sits at the meeting point of algorithm design, complexity theory, system architecture, and hardware development.

Applications of quantum computing come in a plethora of flavors and span the gamut from low-level hardware
problems to higher-level and abstract problems. On the one hand, various phenomena of quantum mechanics are used
to implement physical schemes to control and stabilize quantum units of information—**qubits**—or to synthesize 
a quantum circuit as efficiently as possible. On the other hand, more conceptual problems of quantum computing 
involve the breaking of classical encryption protocols, simulation of quantum systems, or even training of 

> *quantum neural networks*.

Reviewer 2: Worth making this a link to https://pennylane.ai/qml/demos_qml.html?

Reviewer 2: Or could also link to what is qml, maybe that is more appropriate.

Author: The 'what is QML' link makes more sense to me, will include.

Advantage and challenge of quantum computing
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> .. image:: /_static/whatisqc/quantum_computer.svg

Reviewer 2: @ikurecic I've noticed that there are newer versions of the images available from Tarik that don't seem to be updated here yet?

Author: Yep, will explain in detail in Slack.

    :align: right
    :width: 42%
    :target: javascript:void(0);

To explore **what quantum computers can do and how to make them do it**, many of their aspects are researched and applied.
For example, the quantum phenomenon of *superposition* allows for quantum computers to explore many possibilities 

> at once and operate entirely differently form their classical counterparts.

Reviewer: ```suggestion
at once and operate entirely differently from their classical counterparts.
```

Perhaps the primary ambition of quantum computing nowadays is to develop a **scalable, fault-tolerant device** to 
implement any and all quantum algorithms. Research in specialized fields like quantum machine learning and quantum 
chemistry is already underway, but existing quantum hardware is limited to devices that are either small, noisy, 
or non-universal—a computing paradigm known as noisy intermediate-scale quantum, or **NISQ**.

On top of that, real devices that can carry out useful quantum algorithms consist of **many interacting components**, 
which must be able to correct errors that inevitably arise from **device–environment interactions**. Architectures 
where errors can be corrected faster than they occur are referred to as *fault-tolerant*.

The bigger picture: quantum advantage
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. image:: /_static/whatisqc/Borealis_quantum_advantage.svg
    :align: right
    :width: 63%
    :target: javascript:void(0);

Whether practical algorithmic speedups are possible in the NISQ regime remains an open problem, but some instances 
have already been demonstrated of quantum devices solving computational problems that would take classical 

> computers an unfeasible amount of time. Despite this, ****quantum computational advantage is already a reality**, 

Reviewer: ```suggestion
computers an unfeasible amount of time. Despite this, **quantum computational advantage is already a reality**, 
```

> computers an unfeasible amount of time. Despite this, ****quantum computational advantage is already a reality**, 

Reviewer: Something's off about these two sentences. :thinking:.
It seems to say:
- Q. advantage has been recently demonstrated
- Despite this, Q. advantage is already a reality

Would be good to use a transition which conveys more a sense of continuation of the previous thought (something like "Because of this"?), rather than a contradiction ("Despite")

Author: Thanks for spotting it. I'd just remove the 'intro' altogether, and start with "Quantum advantage is already a reality" ...

> with an increasing number of commercial and research organizations announcing their breakthroughs—some even 
> making their devices publicly available for further research.

Reviewer 2: I think there is an opportunity to spread some links here, e.g.,

```suggestion
with an `increasing <https://www.nature.com/articles/s41586-022-04725-x>`__ :doc:`number <tutorial_gbs>` of :doc:`commercial <qsim_beyond_classical>` and research organizations announcing their breakthroughs—some even 
making their devices publicly available for further research.
```

Reviewer 2: the words that should be links in the sentence open to being changed 😆

Reviewer 2: Went through the docs, no other demos/pages come to mind for linking but I might be missing something.

Author: I'll give it an attempt.

fault-tolerant quantum computing for the post-NISQ era.

A multifaceted technology to solve real-world problems
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Few technological advances hold a real potential to let us speed through entire stages of scientific development, 
but quantum computers are likely to be one of the cornerstone technologies of the 21st century, 
**changing the way we do research, protect our data, communicate, and understand the world around us**. 
Initial progress has already shown that—with further improvements to quantum hardware setups—quantum computing 
will be used widely and applied to an ever-growing variety of problems across the globe.

.. figure:: /_static/whatisqc/QC_applications.svg
    :align: center
    :width: 63%
    :target: javascript:void(0);

For example, quantum computers could eventually be used to speedrun the development of new chemical compounds for 
medicine or agriculture, enable a perfectly secure exchange of private messages, optimize and enhance existing 
computational algorithms for image classification, traffic management, or product design, and for 

> **thousands of other uses we haven’t thought of yet**.

Reviewer 2: Good opportunity to link to what is qchem!

PennyLane for quantum computing
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

PennyLane is an **open-source cross-platform Python library** that supports a wide array of tasks in quantum computing, quantum machine learning, and quantum chemistry. Its capabilities for the differentiable programming of quantum computers have been designed to seamlessly integrate with classical machine learning libraries, quantum simulators and hardware, giving users the power to train quantum circuits.

> To find out more, visit the `PennyLane Documentation <https://pennylane.readthedocs.io>`_ or check out the gallery of hands-on :doc:`demonstrations <demonstrations>`.

Reviewer 2: ```suggestion
To find out more, visit the `PennyLane Documentation <https://docs.pennylane.ai>`_ or check out the gallery of hands-on :doc:`demonstrations <demonstrations>`.
```
